Gems/DiffuseProbeGrid/Code/Source/Render/DiffuseProbeGridFeatureProcessor.cpp

```diff
@@ -705,6 +705,22 @@ namespace AZ
 
         void DiffuseProbeGridFeatureProcessor::OnRenderPipelinePassesChanged([[maybe_unused]] RPI::RenderPipeline* renderPipeline)
         {
+            // change the attachment on the AuxGeom pass to use the output of the visualization pass
+            RPI::PassFilter auxGeomPassFilter = RPI::PassFilter::CreateWithPassName(AZ::Name("AuxGeomPass"), renderPipeline);
+            RPI::Pass* auxGeomPass = RPI::PassSystemInterface::Get()->FindFirstPass(auxGeomPassFilter);
+            RPI::PassFilter visualizationPassFilter = RPI::PassFilter::CreateWithPassName(AZ::Name("DiffuseProbeGridVisualizationPass"), renderPipeline);
+            RPI::Pass* visualizationPass = RPI::PassSystemInterface::Get()->FindFirstPass(visualizationPassFilter);
+
+            if (auxGeomPass && visualizationPass && visualizationPass->GetInputOutputCount())
+            {
+                RPI::PassAttachmentBinding& visualizationBinding = visualizationPass->GetInputOutputBinding(0);
+                RPI::PassAttachmentBinding* auxGeomBinding = auxGeomPass->FindAttachmentBinding(AZ::Name("ColorInputOutput"));
+                if (auxGeomBinding)
+                {
+                    auxGeomBinding->SetAttachment(visualizationBinding.GetAttachment());
```

Comment:
Does the DiffuseProbeGridVisualizationPass create a new output color render target or use a color input/output (not create a new one)?

If it doesn't create a new render target, I think you don't need to update the attachment here.

Reply:
Based on previous discussions with Anton, I believe it's best practice to always chain the input/output pass attachments to ensure proper ordering by the frame graph.

Reply:
Maybe using the pipeline attachment would be better. I understand chain the input/output would be safe to do. But update the chain is messy and if any other user insert another pass before AuxGeom pass, then it's tricky to connect their slots.

Reply:
It seems the change here assumes AuxGeom pass is right after PostFX pass, and you add the new pass right after PostFX then you connect the output of the new pass to the auxgeom pass. What if the auxgeom pass is not adjacent to PostFX pass anymore?

Reply:
I think this is a general problem with inserting passes into the pipeline, since it is impossible to predict which passes are present.